x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherPluginTests.java

```diff
@@ -97,4 +100,43 @@ public void testThreadPoolSize() {
                 .build();
         assertThat(Watcher.getWatcherThreadPoolSize(noDataNodeSettings), is(1));
     }
+
+    private class TestNotificationService extends NotificationService<Account> {
+
+        boolean calledCreateAccount = false;
+
+        TestNotificationService(Settings settings, String type) {
+            super(settings, type);
+        }
+
+        @Override
+        protected Account createAccount(String name, Settings accountSettings) {
+            return null;
+        }
+
+        @Override
+        public synchronized void loadSettings(Settings settings) {
+            calledCreateAccount = true;
+            super.loadSettings(settings);
+        }
+    }
+
+    public void testReload() throws Exception {
+        Settings settings = Settings.builder()
+            .put("xpack.watcher.enabled", false)
```

Comment:
watcher is not enabled, but reloading works? That could be a noop in this case, or?

Reply:
alternatively you could mock a NotificationService and check with `verify` if the `loadSettings` method was called, no need to have an impl then

Reply:
in terms of the enable, there is no check to see if the service is active, like there are in things like register components. But there should be, since the case involving a reload for a inactive plugin will attempt to register some things and likely fail miserably. Will add & add tests, and just mock them so i dont have to impl them.

Reply:
b97f21b